### PR TITLE
[WFLY-10611] Elytron custom-security-event-listener test+doc

### PR DESCRIPTION
**Pull request by thaarok**
https://issues.jboss.org/browse/WFLY-10611

needs to be merged in coordination with https://github.com/wildfly/wildfly-core/pull/3285 (WFCORE-3819)
**(tests will be red until ^^ will be merged!)**

Adds tests and documentation required for WFCORE-3819.

### CODE REVIEW

File: testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/audit/CustomSecurityEventListener.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package org.wildfly.test.integration.elytron.audit;
```
Comment (OndrejKotek) [inline testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/audit/CustomSecurityEventListener.java]: Can disclaimers be omitted?

**Comment by OndrejKotek** [comment]
OK for QE.

File: testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/audit/CustomSecurityEventListener.java (hunk 1 above)
Comment (hkalina) [reply testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/audit/CustomSecurityEventListener.java]: fixed

**Comment by msve** [comment]
@hkalina please consider adding link to the new Custom_Components.adoc to the reference list at the bottom of docs/src/main/asciidoc/WildFly_Elytron_Security.adoc

This way the new documentation is not referenced from any top-level document

**Comment by hkalina** [comment]
Fixed.

**Comment by kabir** [comment]
Retest this please

**Comment by kabir** [comment]
@hkalina The errors seem pretty consistent

**Comment by kabir** [comment]
Retest this please

**Pull request merged**